src/frontend/src/pages/ProjectPage/pages/ContractPage/pages/ManagePersonnelPage/AddPersonnelSideSheet/AddPersonnelForm/components/AddPersonnelFormLine.tsx

```diff
@@ -0,0 +1,119 @@
+import * as React from 'react';
+import * as styles from '../../styles.less';
+import Personnel from '../../../../../../../../../models/Personnel';
+import SelectionCell from '../../../components/SelectionCell';
+import AddPersonnelFormTextInput from './AddPersonnelFormTextInput';
+import { SkeletonBar } from '@equinor/fusion-components';
+import AddPersonnelFormDisciplinesDropDown from './AddPersonnelFormDisciplinesDropDown';
+import useForm from '../../../../../../../../../hooks/useForm';
+import { BasePosition } from '@equinor/fusion';
+import { PopOverMenu } from './AddPersonnelFormLinePopOverMenu';
+import PersonnelLine from '../../models/PersonnelLine';
+
+type AddPersonnelFormLineProps = {
+    person: PersonnelLine;
+    setPersonState: (person: PersonnelLine) => void;
+    saveInProgress: boolean;
+    isFetchingBasePositions: Boolean;
+    basePositions: BasePosition[];
+    deletePerson: (person: PersonnelLine) => void;
+};
+
+export const AddPersonnelFormLine: React.FC<AddPersonnelFormLineProps> = ({
+    person,
+    setPersonState,
+    saveInProgress,
+    isFetchingBasePositions,
+    basePositions,
+    deletePerson,
+}) => {
+    const validateForm = React.useCallback((formState: Personnel) => {
+        return Boolean(
+            formState.firstName && formState.lastName && formState.phoneNumber && formState.mail
+        );
+    }, []);
+
+    const { formState, formFieldSetter, setFormState } = useForm(
+        () => person,
+        validateForm,
+        person
+    );
+
+    const onSelect = React.useCallback(() => {
+        formFieldSetter('selected')(!formState?.selected);
+    }, [formState]);
```

Comment:
Not sure about this, but I think formFieldSetter should be in the dependency array here 🤔 I guess if it works the way it is now, the formFieldSetter does not change so it's not needed, but I would check the implementation of useForm to be sure that the formFieldSetter is never set to a new instance of the function

Reply:
there are no dependencies set for formFieldSetter  in useForm. 
Think this should be fine